### PR TITLE
Add KeyValuePair<K, V> to default formatters

### LINKED ISSUE #2244: Add KeyValuePair<TKey, TValue> to the default formatters
Many times the ToString representation of KeyValuePair isn't very helpful so I think it'd make sense if there was a default formatter that formatted the key and value separately using the registered formatters.

### PR DESCRIPTION
**Pull request by mikkelbu**
Also replaced some tabs with spaces and removed trailing spaces.

Fixes #2244

### CODE REVIEW

File: src/NUnitFramework/framework/Constraints/MsgUtils.cs [hunk 1]
```diff
@@ -200,6 +205,42 @@ private static string FormatArray(Array array)
...
+        }
+
+        private static string FormatKeyValuePair(object key, object value)
+        {
+            StringBuilder s = new StringBuilder();
+            s.Append('[');
+            if (key != null)
+            {
+                s.Append(FormatValue(key));
+            }
```
Comment (jnm2) [inline src/NUnitFramework/framework/Constraints/MsgUtils.cs]: I don't think we want the `if`. `FormatValue(null)` returns `MsgUtils.Fmt_Null` which is `"null"`.

File: src/NUnitFramework/framework/Constraints/MsgUtils.cs [hunk 2]
```diff
@@ -200,6 +205,42 @@ private static string FormatArray(Array array)
...
+            s.Append(", ");
+            if (value != null)
+            {
+                s.Append(FormatValue(value));
+            }
```
Comment (jnm2) [inline src/NUnitFramework/framework/Constraints/MsgUtils.cs]: Same [I don't think we want the `if`. `FormatValue(null)` returns `MsgUtils.Fmt_Null` which is `"null"`.]

File: src/NUnitFramework/framework/Constraints/MsgUtils.cs [hunk 3]
```diff
@@ -200,6 +205,42 @@ private static string FormatArray(Array array)
...
+
+            object k = valueType.GetProperty("Key").GetValue(value, null);
+            object v = valueType.GetProperty("Value").GetValue(value, null);
+
+            return FormatKeyValuePair(k, v);
+        }
+
+        private static string FormatKeyValuePair(object key, object value)
+        {
+            StringBuilder s = new StringBuilder();
```
Comment (CharliePoole) [inline src/NUnitFramework/framework/Constraints/MsgUtils.cs]: sb?

File: src/NUnitFramework/framework/Constraints/MsgUtils.cs (hunk 1 above)
Comment (CharliePoole) [inline src/NUnitFramework/framework/Constraints/MsgUtils.cs]: Agree.

File: src/NUnitFramework/framework/Constraints/MsgUtils.cs (hunk 2 above)
Comment (CharliePoole) [inline src/NUnitFramework/framework/Constraints/MsgUtils.cs]: +1

Comment (CharliePoole) [inline src/NUnitFramework/framework/Constraints/MsgUtils.cs]: Once the ifs are gone, you can replace multiple lines with use of ApendFormat.

**Comment by mikkelbu** [comment]
I've simplified the formatting as requested by @jnm2 and combined the two test methods into one.

**Pull request merged**